### PR TITLE
revert: Revert UserPassword (#13680)

### PR DESCRIPTION
**Pull request by zomars**
This reverts commit 0a68c6d

## What does this PR do?

amends #13628 

There was an edge case where a user may not have a password set and use CAL identity provider. Causing to fail setting up a password. This PR amends the logic so we make sure to "upsert" every time instead of just "updating".

### CODE REVIEW

File: apps/web/pages/api/auth/changepw.ts [hunk 1]
```diff
@@ -53,12 +53,16 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   const hashedPassword = await hashPassword(newPassword);
-  await prisma.user.update({
+  await prisma.userPassword.upsert({
     where: {
```
Comment (zomars) [inline apps/web/pages/api/auth/changepw.ts:57]: This is ht main difference from the original PR

File: packages/trpc/server/routers/viewer/auth/changePassword.handler.ts [hunk 2]
```diff
@@ -54,12 +49,16 @@ export const changePasswordHandler = async ({ input, ctx }: ChangePasswordOption
...
+    create: {
+      hash: hashedPassword,
+      userId: user.id,
     },
-    data: {
-      password: hashedPassword,
+    update: {
+      hash: hashedPassword,
     },
   });
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/auth/changePassword.handler.ts:63]: This is the main difference from the original PR

File: packages/trpc/server/routers/viewer/organizations/setPassword.handler.ts [hunk 3]
```diff
@@ -44,12 +45,16 @@ export const setPasswordHandler = async ({ ctx, input }: UpdateOptions) => {
     });
 
   const hashedPassword = await hashPassword(newPassword);
-  await prisma.user.update({
+  await prisma.userPassword.upsert({
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/organizations/setPassword.handler.ts:48]: Same here

**Review by zomars — COMMENTED** [review]
Self review done

**Review by emrysal — APPROVED** [review]
👍  - looks good to me now, trust your diagnosis also

**Pull request merged**